ssr/src/page/post_view/video_iter.rs

```diff
@@ -82,6 +90,86 @@ impl<'a, const AUTH: bool> VideoFetchStream<'a, AUTH> {
         Ok(FetchVideosRes {
             posts_stream: Box::pin(chunk_stream),
             end,
+            res_type: FeedResultType::PostCache,
         })
     }
+
+    pub async fn fetch_post_uids_ml_feed_chunked(
+        &self,
+        chunks: usize,
+        _allow_nsfw: bool,
+        video_queue: Vec<PostDetails>,
+    ) -> Result<FetchVideosRes<'a>, PostViewError> {
+        #[cfg(feature = "hydrate")]
+        {
+            use crate::utils::ml_feed::ml_feed_grpcweb::MLFeed;
+            use leptos::expect_context;
+
+            let user_canister_principal = self.canisters.user_canister();
```

Comment:
As discussed, we'll store user canister in local storage on login.
If its not enabled, you can await `auth_canisters` to get the user_principal instead
It will look like this (I promise it will look better with leptos 0.7 :P):
```rust
let cans_store = auth_canisters_store();
let cans_stream = cans_store.to_stream();
let user_canister = loop {
    let Some(cans) = cans_stream.next().await.flatten() else {
        continue
     };
     cans.user_canister();
};
```
(auth_canisters_store: https://github.com/go-bazzinga/hot-or-not-web-leptos-ssr/blob/main/ssr/src/state/canisters.rs#L288)